drivers/sensor/ite_tach_it8xxx2/tach_ite_it8xxx2.c

```diff
@@ -0,0 +1,256 @@
+ *                            |       +-----+-----+    |   _   _   |<--+
+ *                            |------>| F2TL/MRR  |<-+-|  | |_| |_ |<--+
+ *                            |       +-----------+    +-----------+   |
+ *                            |       capture pulses                  T1B (GPJ3)
+ *                            |       in one second
+ *                            |       period
+ *                            |
+ *
+ * Based on the counter value, we can compute the current RPM of external signal
+ * from encoders.
+ */
+
+#include <device.h>
+#include <drivers/pinmux.h>
+#include <drivers/sensor.h>
+#include <dt-bindings/sensor/it8xxx2_tach.h>
+#include <errno.h>
+#include <soc.h>
+#include <soc_dt.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(tach_ite_it8xxx2, CONFIG_SENSOR_LOG_LEVEL);
+
+/*
+ * NOTE: The PWM output maximum is 324Hz in EC LPM, so if we need fan to work
+ *       then don't let EC enter LPM.
+ */
+#define TACH_FREQ		EC_FREQ
+
+/* Device config */
+struct tach_alt_cfg {
+	/* Pinmux control device structure */
+	const struct device *pinctrls;
+	/* GPIO pin */
+	uint8_t pin;
+	/* Alternate function */
+	uint8_t alt_fun;
+};
+
+struct tach_it8xxx2_config {
+	/* Fan x tachometer LSB reading register */
+	uintptr_t reg_fxtlrr;
+	/* Fan x tachometer MSB reading register */
+	uintptr_t reg_fxtmrr;
+	/* Tachometer switch control register */
+	uintptr_t reg_tswctlr;
+	/* Tachometer data valid bit of tswctlr register */
+	int dvs_bit;
+	/* Tachometer data valid status bit of tswctlr register */
+	int chsel_bit;
+	/* Tachometer alternate configuration list */
+	const struct tach_alt_cfg *alt_list;
+	/* Select channel of tachometer */
+	int channel;
+	/* Number of pulses per round of tachometer's input */
+	int pulses_per_round;
+};
+
+/* Driver data */
+struct tach_it8xxx2_data {
+	/* Captured counts of tachometer */
+	uint32_t capture;
+};
+
+static int tach_ch_valid(const struct device *dev, int tach_ch)
+{
+	const struct tach_it8xxx2_config *const config = dev->config;
+	volatile uint8_t *reg_tswctlr = (uint8_t *)config->reg_tswctlr;
+	int dvs_bit = config->dvs_bit;
+	int chsel_bit = config->chsel_bit;
+	int mask = (dvs_bit | chsel_bit);
+	int valid = 0;
+
+	switch (tach_ch) {
+	case TACH_CHANNEL_A:
+		if ((*reg_tswctlr & mask) == dvs_bit) {
+			valid = 1;
+		}
+		break;
+	case TACH_CHANNEL_B:
+		if ((*reg_tswctlr & mask) == mask) {
+			valid = 1;
+		}
+		break;
+	default:
+		break;
+	}
+
+	return valid;
+}
+
+static int tach_it8xxx2_sample_fetch(const struct device *dev,
+				     enum sensor_channel chan)
+{
+	const struct tach_it8xxx2_config *const config = dev->config;
+	volatile uint8_t *reg_fxtlrr = (uint8_t *)config->reg_fxtlrr;
+	volatile uint8_t *reg_fxtmrr = (uint8_t *)config->reg_fxtmrr;
+	volatile uint8_t *reg_tswctlr = (uint8_t *)config->reg_tswctlr;
+	int tach_ch = config->channel;
+	struct tach_it8xxx2_data *const data = dev->data;
+
+	if (chan != SENSOR_CHAN_RPM) {
```

Comment:
`SENSOR_CHAN_ALL` also needs to be supported in fetch

Reply:
Done